File: tree/Tries.java
package tree;

public class Tries implements ITree<String> {
    private TriesNode root;
    @Override
    public void insert(String word) {
        if(this.root == null){
            this.root = new TriesNode();
        }
        this.root.insert(word.toLowerCase(),0);
    }

    @Override
    public boolean contains(String word) {
        if(this.root == null) return false;
        return this.root.contains(word.toLowerCase(),0);
    }
}

File: src/datastructures/trees/tries/Tries.php
<?php
declare(strict_types=1);
namespace ZimCodes\Self_Learning\DataStructures\Trees\Tries;
class Tries implements ITries{
    private ?INode $root;
    public function __construct(){
        $this->root = null;
    }
    function insert(string $word): void{
        if($this->root === null) {
            $this->root = new Node();
        }
        $this->root->insert($word,0);
    }

    function contains(string $word): bool{
        if($this->root === null) return false;
        return $this->root->contains($word,0);
    }

    function getAllWords(?INode $startNode = null, string $word = "", array &$words = []): array{
        $curNode = $startNode ?? $this->root;
        foreach($curNode->getLetters() as $letter => $node){
            if($letter === '*'){
                $words[] = $word;
            }else{
                $this->getAllWords($node,$word.$letter,$words);
            }
        }
        return $words;
    }
}